### PR TITLE
Avoid chat flash render

### LINKED ISSUE #2385: Chat settings resets when reopened
Found a minor bug on the frontend: when a preset is selected, reopening the settings tab will jump back to the first option.

Thanks!

### PR DESCRIPTION
**Pull request by notmd**
- Fix flash when rendering chat conversation. The logic when render avatar in MessageTableEntry now rely on CSS instead of JS.
 - Fix `ChatConfigSummary` cause hydration error. Sorry about this unrelated change, I though it was an easy fix so I include in this PR but turn out it's quite complex
 - fix #2385
 New UI of ChatMessageEntry
 
image

### CODE REVIEW

File: website/src/components/Chat/ChatSection.tsx [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
+import { useCacheConfig } from "src/hooks/chat/useCacheConfig";
 import { ChatConfigForm } from "src/types/Chat";
 
-import { ChatConfigSummary } from "./ChatConfigSummary";
+import { ChatConfigSaver } from "./ChatConfigSaver";
 import { useChatContext } from "./ChatContext";
 import { ChatConversation } from "./ChatConversation";
 import { InferencePoweredBy } from "./InferencePoweredBy";
 
+const ChatConfigSummary = dynamic(() => import("./ChatConfigSummary"), { ssr: false });
```
Comment (AbdBarho) [inline website/src/components/Chat/ChatSection.tsx:12]: why is this necessary?

File: website/src/pages/chat/[id].tsx [hunk 2]
```diff
@@ -46,12 +48,13 @@ export const getServerSideProps: GetServerSideProps<ChatProps, { id: string }> =
 
   const token = await getToken({ req });
   const client = createInferenceClient(token!);
-  const modelInfos = await client.get_models();
+  const [modelInfos, chat] = await Promise.all([client.get_models(), client.get_chat(query.id as string)]);
```
[the remark this answers is not in the document]
Comment (notmd) [reply website/src/pages/chat/[id].tsx:51]: Fetching on serverside will make the message available at the first time, otherwise, the UI will jump around.

File: website/src/components/Chat/ChatSection.tsx (hunk 1 above)
Comment (notmd) [reply website/src/components/Chat/ChatSection.tsx:12]: The local storage cause a hydration error. I'm can't find a way to make it gratefully hydrate. It's the easiest solution for now. The ChatConfigSummary will be removed on the next UI refactor I think.

File: website/src/components/Chat/ChatConfigDrawer.tsx [hunk 3]
```diff
@@ -114,24 +115,27 @@ const ChatConfigForm = () => {
...
           ? customPresetDefaultValue
           : presets.find((preset) => preset.name === newPresetName)!.sampling_parameters;
-      setPresetName(newPresetName);
+      // setPresetName(newPresetName);
       reset({ ...config, model_config_name: selectedModel });
     },
     [presets, reset, selectedModel]
   );
+
+  const selectedPresetName = (
```
[the remark this answers is not in the document]
Comment (notmd) [reply website/src/components/Chat/ChatConfigDrawer.tsx:135]: I think it will be same, we have to know which preset is selected to highlight the UI anyway.

File: website/src/components/Chat/ChatConfigDrawer.tsx [hunk 4]
```diff
@@ -114,24 +115,27 @@ const ChatConfigForm = () => {
...
-
   const handlePresetChange = useCallback(
     (e: ChangeEvent<HTMLSelectElement>) => {
       const newPresetName = e.target.value;
       const config =
         newPresetName === customPresetName
           ? customPresetDefaultValue
           : presets.find((preset) => preset.name === newPresetName)!.sampling_parameters;
-      setPresetName(newPresetName);
+      // setPresetName(newPresetName);
```
[the remark this answers is not in the document]
Comment (notmd) [reply website/src/components/Chat/ChatConfigDrawer.tsx:129]: Will do in next PR, dont want to deal with rebase/squash stuffs anymore.

**Review by AbdBarho — APPROVED** [review]
We can merge this now, clean up later.
